Cannot build standalone formatter with "OpenJDK 64-Bit Server VM GraalVM CE 22.0.0.2 (build 17.0.2+8-jvmci-22.0-b05)"

The standalone formatter `tvdformat.jar`, version 21.4.2 works with Java 8, 11 and 17.  It requires a JDK 17 to build. Building the executable JAR works with **"Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)"**. But when I try to build it with the GraalVM CE 22.0.0.2 (`mvn clean package`)I get various test errors. Here are some examples:

```
[INFO] Running com.trivadis.plsql.formatter.settings.tests.grammar.Json_functions
[ERROR] Tests run: 4, Failures: 4, Errors: 0, Skipped: 0, Time elapsed: 5.319 s <<< FAILURE! - in com.trivadis.plsql.formatter.settings.tests.grammar.Json_functions
[ERROR] json_value_on_same_line  Time elapsed: 1.359 s  <<< FAILURE!
java.lang.AssertionError: oracle.dbtools.arbori.ScriptException: org.graalvm.polyglot.PolyglotException: java.lang.AssertionError
	at com.trivadis.plsql.formatter.settings.tests.grammar.Json_functions.json_value_on_same_line(Json_functions.java:91)
Caused by: oracle.dbtools.arbori.ScriptException: org.graalvm.polyglot.PolyglotException: java.lang.AssertionError
	at com.trivadis.plsql.formatter.settings.tests.grammar.Json_functions.json_value_on_same_line(Json_functions.java:91)
Caused by: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: java.lang.AssertionError
	at com.trivadis.plsql.formatter.settings.tests.grammar.Json_functions.json_value_on_same_line(Json_functions.java:91)
Caused by: org.graalvm.polyglot.PolyglotException: java.lang.AssertionError
	at com.trivadis.plsql.formatter.settings.tests.grammar.Json_functions.json_value_on_same_line(Json_functions.java:91)
(...)
org.graalvm.js.scriptengine/com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.invokeFunction(GraalJSScriptEngine.java:556)
	at oracle.dbtools.arbori.JSFunc.eval(Predicate.java:968)
	at oracle.dbtools.arbori.CompositeExpr.eval(Predicate.java:1057)
	at oracle.dbtools.arbori.CompositeExpr.eval(Predicate.java:1057)
	at oracle.dbtools.arbori.CompositeExpr.eval(Predicate.java:1059)
(...)
	at oracle.dbtools.arbori.CompositeExpr.eval(Predicate.java:1059)
	at oracle.dbtools.arbori.IndependentAttribute.filterUnaryPredicates(IndependentAttribute.java:235)
	at oracle.dbtools.arbori.IndependentAttribute.addTuples(IndependentAttribute.java:112)
	at oracle.dbtools.arbori.IndependentAttribute.initContent(IndependentAttribute.java:69)
	at oracle.dbtools.arbori.AttributeDefinitions.evalDimensions(AttributeDefinitions.java:232)
	at oracle.dbtools.arbori.Program.evalDimensions(Program.java:1391)
	at oracle.dbtools.arbori.Program$PredicateCmd._eval(Program.java:1483)
	at oracle.dbtools.arbori.Program$PredicateCmd.eval(Program.java:1452)
	at oracle.dbtools.arbori.Program.eval(Program.java:1267)
	at oracle.dbtools.app.Format.format(Format.java:375)
	at <js>.formatFile(<eval>:538)
	at <js>.formatFiles(<eval>:550)
	at <js>.run(<eval>:604)
	at <js>.handleEvent(<eval>:639)
	... 79 more
```

When I run `mvn -DskipTests=true clean package` a working `tvdformat.jar` is produced.

It looks like a testing issue only, but it is not. I cannot produce a native image with GraalVM 22.0.0.2. 

When I run `mvn -Dskip.native=false -DskipTests=true clean package` the following output is produced

```
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------< com.trivadis.plsql.formatter:tvdformat >---------------
[INFO] Building com.trivadis.org.plsql.formatter.settings 21.4.3-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ tvdformat ---
[INFO] Deleting /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ tvdformat ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 7 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ tvdformat ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ tvdformat ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 10 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ tvdformat ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 171 source files to /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ tvdformat ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ tvdformat ---
[INFO] Building jar: /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-shade-plugin:3.2.4:shade (default) @ tvdformat ---
[INFO] Including org.graalvm.js:js:jar:22.0.0.2 in the shaded jar.
[INFO] Including org.graalvm.regex:regex:jar:22.0.0.2 in the shaded jar.
[INFO] Including org.graalvm.truffle:truffle-api:jar:22.0.0.2 in the shaded jar.
[INFO] Including org.graalvm.sdk:graal-sdk:jar:22.0.0.2 in the shaded jar.
[INFO] Including org.graalvm.js:js-scriptengine:jar:22.0.0.2 in the shaded jar.
[INFO] Including com.google.code.findbugs:jsr305:jar:3.0.2 in the shaded jar.
[INFO] Including oracle.dbtools:dbtools-common:jar:21.4.1 in the shaded jar.
[INFO] Including org.reflections:reflections:jar:0.10.2 in the shaded jar.
[INFO] Including org.javassist:javassist:jar:3.28.0-GA in the shaded jar.
[INFO] Including org.slf4j:slf4j-api:jar:1.7.32 in the shaded jar.
[INFO] Including org.slf4j:slf4j-jdk14:jar:1.7.32 in the shaded jar.
[INFO] Replacing /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat.jar with /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT-shaded.jar
[INFO] 
[INFO] --- native-maven-plugin:0.9.9:build (build-native) @ tvdformat ---
[INFO] Executing: /Users/phs/Applications/graalvm-ce-java17-22.0.0.2/Contents/Home/bin/native-image -cp /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT.jar:/Users/phs/.m2/repository/oracle/dbtools/dbtools-common/21.4.1/dbtools-common-21.4.1.jar:/Users/phs/.m2/repository/org/reflections/reflections/0.10.2/reflections-0.10.2.jar:/Users/phs/.m2/repository/org/javassist/javassist/3.28.0-GA/javassist-3.28.0-GA.jar:/Users/phs/.m2/repository/org/slf4j/slf4j-api/1.7.32/slf4j-api-1.7.32.jar:/Users/phs/.m2/repository/org/slf4j/slf4j-jdk14/1.7.32/slf4j-jdk14-1.7.32.jar --features=com.trivadis.plsql.formatter.RuntimeReflectionRegistrationFeature --language:js --no-fallback -H:+ReportExceptionStackTraces -H:Class=com.trivadis.plsql.formatter.TvdFormat -H:Name=tvdformat
========================================================================================================================
GraalVM Native Image: Generating 'tvdformat'...
========================================================================================================================
[1/7] Initializing...
                                                                                    (6.4s @ 0.13GB)
Error: Error parsing JNI configuration in jar:file:/Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT.jar!/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/jni-config.json:
Could not resolve org.graalvm.nativebridge.jni.JNIExceptionWrapperEntryPoints for reflection configuration. Reason: java.lang.ClassNotFoundException: org.graalvm.nativebridge.jni.JNIExceptionWrapperEntryPoints. To allow unresolvable reflection configuration, use option --allow-incomplete-classpath
Verify that the configuration matches the schema described in the -H:PrintFlags=+ output for option JNIConfigurationResources.
com.oracle.svm.core.util.UserError$UserException: Error parsing JNI configuration in jar:file:/Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT.jar!/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/jni-config.json:
Could not resolve org.graalvm.nativebridge.jni.JNIExceptionWrapperEntryPoints for reflection configuration. Reason: java.lang.ClassNotFoundException: org.graalvm.nativebridge.jni.JNIExceptionWrapperEntryPoints. To allow unresolvable reflection configuration, use option --allow-incomplete-classpath
Verify that the configuration matches the schema described in the -H:PrintFlags=+ output for option JNIConfigurationResources.
	at com.oracle.svm.core.util.UserError.abort(UserError.java:73)
	at com.oracle.svm.hosted.config.ConfigurationParserUtils.doParseAndRegister(ConfigurationParserUtils.java:135)
	at com.oracle.svm.hosted.config.ConfigurationParserUtils.lambda$parseAndRegisterConfigurations$3(ConfigurationParserUtils.java:113)
	at java.base/java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:214)
	at com.oracle.svm.hosted.config.ConfigurationParserUtils$1.tryAdvance(ConfigurationParserUtils.java:106)
	at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:332)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:310)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:515)
	at java.base/java.util.stream.IntPipeline.sum(IntPipeline.java:473)
	at com.oracle.svm.hosted.config.ConfigurationParserUtils.parseAndRegisterConfigurations(ConfigurationParserUtils.java:115)
	at com.oracle.svm.jni.access.JNIAccessFeature.afterRegistration(JNIAccessFeature.java:128)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$setupNativeImage$14(NativeImageGenerator.java:802)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:74)
	at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:802)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:534)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:494)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:426)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:587)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:126)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:617)
------------------------------------------------------------------------------------------------------------------------
                         0.2s (3.4% of total time) in 8 GCs | Peak RSS: 0.67GB | CPU load: 3.35
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat.build_artifacts.txt
========================================================================================================================
Failed generating 'tvdformat' after 3.5s.
Error: Image build request failed with exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  16.784 s
[INFO] Finished at: 2022-01-29T17:29:36+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.graalvm.buildtools:native-maven-plugin:0.9.9:build (build-native) on project tvdformat: Execution of /Users/phs/Applications/graalvm-ce-java17-22.0.0.2/Contents/Home/bin/native-image -cp /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT.jar:/Users/phs/.m2/repository/oracle/dbtools/dbtools-common/21.4.1/dbtools-common-21.4.1.jar:/Users/phs/.m2/repository/org/reflections/reflections/0.10.2/reflections-0.10.2.jar:/Users/phs/.m2/repository/org/javassist/javassist/3.28.0-GA/javassist-3.28.0-GA.jar:/Users/phs/.m2/repository/org/slf4j/slf4j-api/1.7.32/slf4j-api-1.7.32.jar:/Users/phs/.m2/repository/org/slf4j/slf4j-jdk14/1.7.32/slf4j-jdk14-1.7.32.jar --features=com.trivadis.plsql.formatter.RuntimeReflectionRegistrationFeature --language:js --no-fallback -H:+ReportExceptionStackTraces -H:Class=com.trivadis.plsql.formatter.TvdFormat -H:Name=tvdformat returned non-zero result -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

It works when using **"OpenJDK 64-Bit Server VM GraalVM CE 21.3.0 (build 17.0.1+12-jvmci-21.3-b05, mixed mode, sharing)"**. Here's the complete output of `mvn -Dskip.native=false -DskipTests=true clean package`:

```
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------< com.trivadis.plsql.formatter:tvdformat >---------------
[INFO] Building com.trivadis.org.plsql.formatter.settings 21.4.3-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ tvdformat ---
[INFO] Deleting /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ tvdformat ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 7 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ tvdformat ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ tvdformat ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 10 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ tvdformat ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 171 source files to /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ tvdformat ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ tvdformat ---
[INFO] Building jar: /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-shade-plugin:3.2.4:shade (default) @ tvdformat ---
[INFO] Including org.graalvm.js:js:jar:22.0.0.2 in the shaded jar.
[INFO] Including org.graalvm.regex:regex:jar:22.0.0.2 in the shaded jar.
[INFO] Including org.graalvm.truffle:truffle-api:jar:22.0.0.2 in the shaded jar.
[INFO] Including org.graalvm.sdk:graal-sdk:jar:22.0.0.2 in the shaded jar.
[INFO] Including org.graalvm.js:js-scriptengine:jar:22.0.0.2 in the shaded jar.
[INFO] Including com.google.code.findbugs:jsr305:jar:3.0.2 in the shaded jar.
[INFO] Including oracle.dbtools:dbtools-common:jar:21.4.1 in the shaded jar.
[INFO] Including org.reflections:reflections:jar:0.10.2 in the shaded jar.
[INFO] Including org.javassist:javassist:jar:3.28.0-GA in the shaded jar.
[INFO] Including org.slf4j:slf4j-api:jar:1.7.32 in the shaded jar.
[INFO] Including org.slf4j:slf4j-jdk14:jar:1.7.32 in the shaded jar.
[INFO] Replacing /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat.jar with /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT-shaded.jar
[INFO] 
[INFO] --- native-maven-plugin:0.9.9:build (build-native) @ tvdformat ---
[INFO] Executing: /Users/phs/Applications/graalvm-ce-java17-21.3.0/Contents/Home/bin/native-image -cp /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat-21.4.3-SNAPSHOT.jar:/Users/phs/.m2/repository/oracle/dbtools/dbtools-common/21.4.1/dbtools-common-21.4.1.jar:/Users/phs/.m2/repository/org/reflections/reflections/0.10.2/reflections-0.10.2.jar:/Users/phs/.m2/repository/org/javassist/javassist/3.28.0-GA/javassist-3.28.0-GA.jar:/Users/phs/.m2/repository/org/slf4j/slf4j-api/1.7.32/slf4j-api-1.7.32.jar:/Users/phs/.m2/repository/org/slf4j/slf4j-jdk14/1.7.32/slf4j-jdk14-1.7.32.jar --features=com.trivadis.plsql.formatter.RuntimeReflectionRegistrationFeature --language:js --no-fallback -H:+ReportExceptionStackTraces -H:Class=com.trivadis.plsql.formatter.TvdFormat -H:Name=tvdformat
[tvdformat:52150]    classlist:   3,374.18 ms,  0.94 GB
[tvdformat:52150]        (cap):   2,347.33 ms,  0.94 GB
[tvdformat:52150]        setup:   6,760.01 ms,  0.94 GB
Jan 29, 2022 5:35:35 PM org.reflections.Reflections scan
INFO: Reflections took 154 ms to scan 1 urls, producing 29 keys and 90 values
Jan 29, 2022 5:35:35 PM org.reflections.Reflections scan
INFO: Reflections took 43 ms to scan 1 urls, producing 13 keys and 41 values
Jan 29, 2022 5:35:35 PM org.reflections.Reflections scan
INFO: Reflections took 96 ms to scan 1 urls, producing 26 keys and 61 values
Jan 29, 2022 5:35:35 PM org.reflections.Reflections scan
INFO: Reflections took 618 ms to scan 1 urls, producing 217 keys and 1029 values
Jan 29, 2022 5:35:35 PM org.reflections.Reflections scan
INFO: Reflections took 39 ms to scan 1 urls, producing 24 keys and 39 values
[tvdformat:52150]     (clinit):   1,474.15 ms,  6.17 GB
[tvdformat:52150]   (typeflow):  17,653.56 ms,  6.17 GB
[tvdformat:52150]    (objects):  40,138.07 ms,  6.17 GB
[tvdformat:52150]   (features):  17,699.81 ms,  6.17 GB
[tvdformat:52150]     analysis:  79,682.91 ms,  6.17 GB
[tvdformat:52150]     universe:   4,141.25 ms,  6.17 GB
10971 method(s) included for runtime compilation
[tvdformat:52150]      (parse):  12,428.48 ms,  6.29 GB
[tvdformat:52150]     (inline):   7,662.57 ms,  6.97 GB
[tvdformat:52150]    (compile):  41,592.58 ms,  6.77 GB
[tvdformat:52150]      compile:  67,028.94 ms,  6.72 GB
[tvdformat:52150]        image:   7,322.17 ms,  6.80 GB
[tvdformat:52150]        write:   3,395.14 ms,  6.80 GB
[tvdformat:52150]      [total]: 177,198.60 ms,  6.80 GB
# Printing build artifacts to: /Users/phs/github/trivadis/plsql-formatter-settings/standalone/target/tvdformat.build_artifacts.txt
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:11 min
[INFO] Finished at: 2022-01-29T17:38:21+01:00
[INFO] ------------------------------------------------------------------------
```

--- a/standalone/README.md
+++ b/standalone/README.md
@@ -11,13 +11,13 @@ This project contains JUnit tests for
 - the SQLcl command `tvdformat`
 - the standalone exectable `tvdformat` 
 
-The project requires a JDK 17, but it produces a Java 8 executable JAR file. A GraalVM JDK is required only if you want to build a [native image](https://www.graalvm.org/reference-manual/native-image/). GraalVM 22.0.0.2 does not work in with the libraries from SQLcl 21.4.1 (throws `org.graalvm.polyglot.PolyglotException: java.lang.AssertionError`). Please use GraalVM 21.3.0 instead. 
+The project requires a JDK 17, but it produces a Java 11 executable JAR file. A GraalVM JDK is required only if you want to build a [native image](https://www.graalvm.org/reference-manual/native-image/).
 
 ## Running the Standalone Formatter
 
 ### Executable JAR
 
-The `tvdformat.jar` is a shaded, executable JAR that is part of a [release](https://github.com/Trivadis/plsql-formatter-settings/releases). It contains all dependent Java classes and needs a JDK 8 or higher at runtime.
+The `tvdformat.jar` is a shaded, executable JAR that is part of a [release](https://github.com/Trivadis/plsql-formatter-settings/releases). It contains all dependent Java classes and needs a JDK 11 or higher at runtime.
 
 To run it, open a terminal window and type
 
@@ -33,11 +33,11 @@ A native image is a platform specific executable. It does not require a JDK at r
 
 OS      | amd64 (Intel))? | aarch64 (ARM)? |
 ------- | :-------------: | :------------: |
-macOS   | yes             | -              |
+macOS   | yes             | yes            |
 Linux   | yes             | yes            |
 Windows | yes             | -              |
 
-Currently there is no way to produce an ARM based (aarch64) native image for macOS and Windows. 
+Currently there is no way to produce an ARM based (aarch64) native image for Windows. 
 
 Native images are not part of a release. You have to build them yourself as described [below](#how-to-build).
 
@@ -51,10 +51,10 @@ The parameters are the same as for the [executable JAR](#executable-jar).
 
 ## How to Build
 
-1. [Download](https://www.oracle.com/tools/downloads/sqlcl-downloads.html) and install SQLcl 21.4.0
-2. [Download](https://github.com/graalvm/graalvm-ce-builds/releases) and install the GraalVM JDK 17 21.3.0
+1. [Download](https://www.oracle.com/tools/downloads/sqlcl-downloads.html) and install SQLcl 22.2.0
+2. [Download](https://github.com/graalvm/graalvm-ce-builds/releases) and install the GraalVM JDK 17 22.1.0
 3. Go to the `bin` directory of the GraalVM JDK and run `./gu install native-image`, if you want to produce a native image
-4. [Download](https://maven.apache.org/download.cgi) and install Apache Maven 3.8.4
+4. [Download](https://maven.apache.org/download.cgi) and install Apache Maven 3.8.6
 5. [Download](https://git-scm.com/downloads) and install a git command line client
 6. Clone the plsql-formatter-settings repository
 7. Open a terminal window in the plsql-formatter-settings root folder and type

--- a/standalone/pom.xml
+++ b/standalone/pom.xml
@@ -354,7 +354,6 @@
                                 <configuration>
                                     <buildArgs combine.children="append">
                                         <buildArg>-H:IncludeResources=.*</buildArg>
-                                        <buildArg>--allow-incomplete-classpath</buildArg>
                                     </buildArgs>
                                 </configuration>
                             </execution>

--- a/standalone/src/main/resources/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/jni-config.json
+++ b/standalone/src/main/resources/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/jni-config.json
@@ -11,6 +11,17 @@
     ]
   },
   {
+    "name": "java.lang.Boolean",
+    "methods": [
+      {
+        "name": "getBoolean",
+        "parameterTypes": [
+          "java.lang.String"
+        ]
+      }
+    ]
+  },
+  {
     "name": "java.lang.ClassLoader",
     "methods": [
       {
@@ -62,16 +73,5 @@
   },
   {
     "name": "jdk.internal.loader.ClassLoaders$PlatformClassLoader"
-  },
-  {
-    "name": "org.graalvm.nativebridge.jni.JNIExceptionWrapperEntryPoints",
-    "methods": [
-      {
-        "name": "getClassName",
-        "parameterTypes": [
-          "java.lang.Class"
-        ]
-      }
-    ]
   }
 ]

--- a/standalone/src/main/resources/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/reflect-config.json
+++ b/standalone/src/main/resources/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/reflect-config.json
@@ -1,6 +1,6 @@
 [
   {
-    "name" : "com.trivadis.plsql.formatter.ScriptRunnerContext",
+    "name": "com.trivadis.plsql.formatter.ScriptRunnerContext",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -11,7 +11,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.io.BufferedWriter",
+    "name": "java.io.BufferedWriter",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -22,7 +22,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.io.File",
+    "name": "java.io.File",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -33,7 +33,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.io.FileInputStream",
+    "name": "java.io.FileInputStream",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -44,7 +44,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.io.PrintStream",
+    "name": "java.io.PrintStream",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -55,13 +55,15 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.lang.Class",
-    "methods" : [
-      {"name" : "getDeclaredField"}
+    "name": "java.lang.Class",
+    "methods": [
+      {
+        "name": "getDeclaredField"
+      }
     ]
   },
   {
-    "name" : "java.lang.Integer",
+    "name": "java.lang.Integer",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -72,14 +74,18 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.lang.reflect.Field",
-    "methods" : [
-      { "name" : "setAccessible"},
-      { "name" : "get"}
+    "name": "java.lang.reflect.Field",
+    "methods": [
+      {
+        "name": "setAccessible"
+      },
+      {
+        "name": "get"
+      }
     ]
   },
   {
-    "name" : "java.lang.String",
+    "name": "java.lang.String",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -90,7 +96,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.lang.StringBuilder",
+    "name": "java.lang.StringBuilder",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -101,14 +107,14 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.lang.System",
+    "name": "java.lang.System",
     "allPublicConstructors": true,
     "allPublicMethods": true,
     "allPublicFields": true,
     "allPublicClasses": true
   },
   {
-    "name" : "java.net.URI",
+    "name": "java.net.URI",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -119,7 +125,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.nio.file.Files",
+    "name": "java.nio.file.Files",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -130,7 +136,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.nio.file.FileSystems",
+    "name": "java.nio.file.FileSystems",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -141,7 +147,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.nio.file.Paths",
+    "name": "java.nio.file.Paths",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -152,7 +158,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.ArrayList",
+    "name": "java.util.ArrayList",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -163,7 +169,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.Arrays",
+    "name": "java.util.Arrays",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -174,7 +180,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.Collection",
+    "name": "java.util.Collection",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -185,7 +191,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.HashMap",
+    "name": "java.util.HashMap",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -196,7 +202,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.HashMap$KeySet",
+    "name": "java.util.HashMap$KeySet",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -207,7 +213,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.HashSet",
+    "name": "java.util.HashSet",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -218,7 +224,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.Iterator",
+    "name": "java.util.Iterator",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -229,7 +235,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.logging.LogManager",
+    "name": "java.util.logging.LogManager",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -240,7 +246,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.regex.Matcher",
+    "name": "java.util.regex.Matcher",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -251,7 +257,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.regex.Pattern",
+    "name": "java.util.regex.Pattern",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -262,7 +268,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.stream.Collectors",
+    "name": "java.util.stream.Collectors",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -273,7 +279,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.stream.ReferencePipeline$Head",
+    "name": "java.util.stream.ReferencePipeline$Head",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -284,7 +290,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.stream.Stream",
+    "name": "java.util.stream.Stream",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -295,7 +301,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "java.util.TreeSet",
+    "name": "java.util.TreeSet",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -306,7 +312,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.app.Format",
+    "name": "oracle.dbtools.app.Format",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -317,7 +323,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.app.Format$Breaks",
+    "name": "oracle.dbtools.app.Format$Breaks",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -328,7 +334,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.app.Format$BreaksX2",
+    "name": "oracle.dbtools.app.Format$BreaksX2",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -339,7 +345,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.app.Format$Case",
+    "name": "oracle.dbtools.app.Format$Case",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -350,7 +356,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.app.Format$FlowControl",
+    "name": "oracle.dbtools.app.Format$FlowControl",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -361,7 +367,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.app.Format$InlineComments",
+    "name": "oracle.dbtools.app.Format$InlineComments",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -372,7 +378,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.app.Format$Space",
+    "name": "oracle.dbtools.app.Format$Space",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -383,7 +389,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.app.Persist2XML",
+    "name": "oracle.dbtools.app.Persist2XML",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -394,7 +400,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.arbori.Program",
+    "name": "oracle.dbtools.arbori.Program",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -405,7 +411,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.arbori.SqlProgram",
+    "name": "oracle.dbtools.arbori.SqlProgram",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -416,7 +422,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.parser.Lexer",
+    "name": "oracle.dbtools.parser.Lexer",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -427,7 +433,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.parser.LexerToken",
+    "name": "oracle.dbtools.parser.LexerToken",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -438,7 +444,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.parser.Parsed",
+    "name": "oracle.dbtools.parser.Parsed",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -449,7 +455,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.parser.ParseNode",
+    "name": "oracle.dbtools.parser.ParseNode",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -460,7 +466,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.parser.plsql.SqlEarley",
+    "name": "oracle.dbtools.parser.plsql.SqlEarley",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -471,7 +477,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.parser.plsql.SyntaxError",
+    "name": "oracle.dbtools.parser.plsql.SyntaxError",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -482,7 +488,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.parser.Substitutions",
+    "name": "oracle.dbtools.parser.Substitutions",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -493,7 +499,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.parser.Token",
+    "name": "oracle.dbtools.parser.Token",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -504,7 +510,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.util.Logger",
+    "name": "oracle.dbtools.util.Logger",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -515,7 +521,7 @@
     "allPublicClasses": true
   },
   {
-    "name" : "oracle.dbtools.util.Service",
+    "name": "oracle.dbtools.util.Service",
     "allDeclaredConstructors": true,
     "allDeclaredMethods": true,
     "allDeclaredFields": true,
@@ -524,5 +530,17 @@
     "allPublicMethods": true,
     "allPublicFields": true,
     "allPublicClasses": true
+  },
+  {
+    "name": "org.graalvm.polyglot.Engine",
+    "methods": [
+      {
+        "name": "getImpl",
+        "parameterTypes": []
+      }
+    ]
+  },
+  {
+    "name": "org.graalvm.polyglot.management.Management"
   }
 ]

--- a/standalone/src/main/resources/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/resource-config.json
+++ b/standalone/src/main/resources/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/resource-config.json
@@ -26,6 +26,9 @@
         "pattern": "\\QMETA-INF/services/org.graalvm.polyglot.impl.AbstractPolyglotImpl\\E"
       },
       {
+        "pattern": "\\Qcom/oracle/truffle/llvm/nfi/SulongNFI.class\\E"
+      },
+      {
         "pattern": "\\Qcom/oracle/truffle/nfi/backend/libffi/LibFFILanguage.class\\E"
       },
       {
@@ -59,7 +62,10 @@
   },
   "bundles": [
     {
-      "name": "oracle.dbtools.util.Messages"
+      "name": "oracle.dbtools.util.Messages",
+      "locales": [
+        ""
+      ]
     }
   ]
 }

--- a/standalone/src/main/resources/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/serialization-config.json
+++ b/standalone/src/main/resources/META-INF/native-image/com.trivadis.plsql.formatter/tvdformat/serialization-config.json
@@ -1,1 +1,6 @@
-[]
+{
+  "types": [
+  ],
+  "lambdaCapturingTypes": [
+  ]
+}

